presto-main/src/main/java/com/facebook/presto/cost/PlanNodeStatsEstimateMath.java

```diff
@@ -259,11 +290,17 @@ private static VariableStatsEstimate addColumnStats(
 
         // FIXME, weights to average. left and right should be equal in most cases anyway
         double newAverageRowSize = newNonNullsRowCount == 0 ? 0 : ((totalSizeLeft + totalSizeRight) / newNonNullsRowCount);
-
-        return VariableStatsEstimate.builder()
+        VariableStatsEstimate.Builder statistics = VariableStatsEstimate.builder()
                 .setStatisticsRange(sum)
                 .setAverageRowSize(newAverageRowSize)
-                .setNullsFraction(newNullsFraction)
-                .build();
+                .setNullsFraction(newNullsFraction);
+        if (shouldUseHistograms) {
+            Optional<ConnectorHistogram> newHistogram = RangeAdditionStrategy.INTERSECT.equals(strategy) ?
```

Comment:
use == comparison for enums so we get compiler errors if we're accidentally comparing with the wrong type of thing.

Reply:
thanks, fixed